### ISSUE TITLE
bug(node): Panic in async function

### ISSUE THREAD

**Issue by ahaapple** [lancedb/lancedb#1505]
### LanceDB version

0.8.0 @lancedb/lancedb

### What happened?

```
2024-08-06T03:01:06.271 app[7811097a9057d8] sjc [info] explicit panic

2024-08-06T03:01:06.281 app[7811097a9057d8] sjc [info] thread 'tokio-runtime-worker' panicked at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/lance-file-0.15.0/src/v2/reader.rs:374:58:

2024-08-06T03:01:06.281 app[7811097a9057d8] sjc [info] explicit panic

2024-08-06T03:01:06.281 app[7811097a9057d8] sjc [info] thread 'tokio-runtime-worker' panicked at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/lance-0.15.0/src/io/exec/scan.rs:199:40:

2024-08-06T03:01:06.281 app[7811097a9057d8] sjc [info] called `Result::unwrap()` on an `Err` value: JoinError::Panic(Id(36939), ...)

2024-08-06T03:01:06.281 app[7811097a9057d8] sjc [info] Failed after 3 attempts: error: Panic in async function
```

**Comment by wjones127** [comment]
Hello, did you enable the new file format? (Passing `useLegacyFormat: false`?)

Not sure how this happened, but seems like you did and it hit this panic here:

https://github.com/lancedb/lance/blob/ef49f80ec501f8b0aa89b11dd7d7064c1387b5c8/rust/lance-file/src/v2/reader.rs#L374

@westonpace  This is from Lance v0.15.0. You can see more of their use case here: https://github.com/memfreeme/memfree/blob/main/vector/db.ts#L14 Any known way this panic could happen?

**Comment by ahaapple** [comment]
I just upgraded to 0.8.0. 
No new parameters were passed and the code did not change. 
So I don't think I have enabled the new file format.

**Comment by westonpace** [comment]
Hmm, there was a bug in the python async where the v2 format was enabled by default.  Maybe the same thing is happening for node.  I'll take a look.

I'm not sure how that panic could happen.  Maybe if reading a file created by a very old version of V2 but I think we've always set the encoding location.

**Comment by westonpace** [comment]
Yeah, it looks like the same bug where a v2 table would be created in the nodejs API if you create an empty table and then add data to it (versus creating a table with initial data).

@ahaapple how much data do you have?  Is it possible to recreate the data?  Otherwise if you can send a sample file I can look into what would be required to migrate it.

**Comment by ahaapple** [comment]
I want to confirm. Does it mean that if I create a table with version 0.5.2 and then insert new data with version 0.8.0, I will still have this problem?

```

async function getTable(db: any, tableName: string): Promise<lancedb.Table> {
if ((await db.tableNames()).includes(tableName)) {
return await db.openTable(tableName);
} else {
return await db.createEmptyTable(tableName, schema);
}
}

```

**Comment by ahaapple** [comment]
I can recreate the data, but there are more than 100 tables and the rebuilding process will take a long time.   I could send you the sample file.

**Comment by westonpace** [comment]
> I want to confirm. Does it mean that if I create a table with version 0.5.2 and then insert new data with version 0.8.0, I will have this issue?

Yes, you would still have the issue on any table created with 0.5.2.

> I can recreate the data, but there are more than 100 tables and the rebuilding process will take a long time. I could send you the sample file.

Ok, I'll look this week if you can get it soon and let you know if I'm able to handle the migration by next week

**Comment by ahaapple** [comment]
Hi, how can I send you the sample file? 
I want to confirm, besides rebuilding the table and waiting for your migration solution, is there any other workaround?

**Comment by westonpace** [comment]
> Hi, how can I send you the sample file?

Attaching any of the files in the `data` directory of one of these datasets will be fine.  If you don't want to attach it to Github (for privacy reasons) you can send it to contact@lancedb.com (just make sure the mention this issue).

> I want to confirm, besides rebuilding the table and waiting for your migration solution, is there any other workaround?

Those are the two solutions:
* Rebuild the table (no fixes needed to do this today)
* Wait for a migration fix

> Hi，In addition, I want to confirm that if I use your enterprise version, can I avoid similar problems? Ensure that there are no errors in data writing and querying

This was a bug (v2 should never have been the default and if we knew it was we would not have made any breaking format changes).  This particular bug wasn't exposed via the enterprise version, but I can't promise the enterprise version will never have a bug.

**Comment by ahaapple** [comment]
Thank you very much for your advice and help

Today is a painful and tangled day for me.

I spent half a day considering whether to migrate vector db to other serverless vector db. In the end, I chose to continue to use lancedb because I like lancedb's natural serverless and its one user one db (one table) architecture.

I have deleted all existing tables today and rebuilt them.

I hope this is the last time I rebuild all user tables. This time I was careless. I know that the first release of almost all databases is unstable, but I still did not do sufficient testing. After that, I will do sufficient testing for my scenario when I upgrade.

**Comment by ahaapple** [comment]
```
rror: lance error: LanceError(IO): Execution error: Row ids did not arrive in sorted order: integers are ordered up to the 0th element, /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/lance-datafusion-0.15.0/src/chunker.rs:47:46
```

In addition, I found that compaction and insert still cannot be done in parallel, and this error will still be reported in version 0.8.0. I am not sure whether this is in line with your expectations.

**Comment by Xuanwo** [comment]
Hi, this issue is invalid because it is already fixed: the bug where empty tables were created with the experimental v2 format by default was fixed in v0.8.0 (confirmed by maintainer), and a migration script was provided for existing tables. I'm closing this now. Feel free to create a new issue if the problem still exists.
